client/client_proxy.go

```diff
@@ -53,24 +54,39 @@ func (c *MockLogClient) GetInclusionProof(ctx context.Context, in *trillian.GetI
 // GetInclusionProofByHash forwards requests.
 func (c *MockLogClient) GetInclusionProofByHash(ctx context.Context, in *trillian.GetInclusionProofByHashRequest, opts ...grpc.CallOption) (*trillian.GetInclusionProofByHashResponse, error) {
 	resp, err := c.c.GetInclusionProofByHash(ctx, in)
+	if err != nil {
+		return nil, err
+	}
 	if c.mGetInclusionProof {
 		h := rand.Intn(len(resp.Proof))
+		if len(resp.GetProof()[h].Hashes) == 0 {
+			log.Printf("Inclusion proof not modified because treesize = 0")
+			return resp, nil
+		}
 		i := rand.Intn(len(resp.Proof[h].Hashes))
 		j := rand.Intn(len(resp.Proof[h].Hashes[i]))
-		resp.Proof[h].Hashes[i][j] ^= 4
+		resp.GetProof()[h].Hashes[i][j] ^= 4
 	}
-	return resp, err
+	return resp, nil
 }
 
 // GetConsistencyProof forwards requests and modifies responses.
 func (c *MockLogClient) GetConsistencyProof(ctx context.Context, in *trillian.GetConsistencyProofRequest, opts ...grpc.CallOption) (*trillian.GetConsistencyProofResponse, error) {
 	resp, err := c.c.GetConsistencyProof(ctx, in)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("consistency: %v", resp.GetProof())
```

Comment:
This is probably too spammy to leave in. Maybe `glog.V(2)` if you really want to keep it?